gluten-core/src/main/scala/io/glutenproject/execution/HashAggregateExecBaseTransformer.scala

```diff
@@ -73,6 +74,24 @@ abstract class HashAggregateExecBaseTransformer(
       aggregateAttributes)
   }
 
+  protected def isGroupingKeysPreGrouped: Boolean = {
+    if (!conf.getConf(GlutenConfig.COLUMNAR_PREFER_STREAMING_AGGREGATE)) {
+      return false
+    }
+    if (groupingExpressions.isEmpty) {
+      return false
+    }
+
+    val childOrdering = child match {
+      case agg: HashAggregateExecBaseTransformer
+          if agg.groupingExpressions == this.groupingExpressions =>
+        agg.child.outputOrdering
```

Comment:
Why propagating child's child's ordering here? It would be great if we could also leave some comments in code. Thanks.

Reply:
added some comments, hope it is clear now